packages/magento-customer/components/NameFields/NameFields.tsx

```diff
@@ -1,35 +1,35 @@
 import { TextFieldElement } from '@graphcommerce/ecommerce-ui'
 import { FormRow, InputCheckmark } from '@graphcommerce/next-ui'
-import { assertFormGqlOperation, Controller, UseFormReturn } from '@graphcommerce/react-hook-form'
+import { assertFormGqlOperation, Controller, useFormContext } from '@graphcommerce/react-hook-form'
 import { i18n } from '@lingui/core'
 import { Trans } from '@lingui/react'
 // eslint-disable-next-line @typescript-eslint/no-restricted-imports
 import { MenuItem, TextField } from '@mui/material'
-import React from 'react'
 
 type NameFieldValues = {
   firstname?: string
   lastname?: string
   prefix?: string
 }
 
-type NameFieldProps = {
-  // eslint-disable-next-line @typescript-eslint/no-explicit-any
-  form: UseFormReturn<any>
-  readOnly?: boolean
+type NameFieldsProps = {
   prefix?: boolean
-  prefixes?: string[]
 }
 
-export function NameFields(props: NameFieldProps) {
+export function NameFields({ prefix = false }: NameFieldsProps) {
   const mr = i18n._(/* i18n */ 'Mr')
   const mrs = i18n._(/* i18n */ 'Mrs')
   const other = i18n._(/* i18n */ 'Other')
 
-  const { prefix, form, readOnly, prefixes = [mr, mrs, other] } = props
+  const form = useFormContext()
+
+  const prefixes = [mr, mrs, other]
+
```

Comment:
Why would you no longer want to be able to pass prefixes in props?

Reply:
Fixed